packages/common/driver-definitions/src/storage.ts

```diff
@@ -450,3 +453,42 @@ export enum FetchSource {
 	default = "default",
 	noCache = "noCache",
 }
+
+/**
+ * This represents the snapshot tree structure with group IDs. It could be a partial snapshot tree
+ * where the missing tree structure inside another tree could be represented with a groupId and the
+ * actual tree structure could be missing. This group Ids could be then be used to fetch the actual
+ * tree structure from the storage service.
+ * @alpha
+ */
+export interface ISnapshotTree2 extends ISnapshotTree {
+	/**
+	 * Represents the group to which the children belongs to.
+	 */
+	groupID?: string;
+
+	trees: { [path: string]: ISnapshotTree2 };
+}
+
+/**
+ * This represents the complete snapshot along with contents. The snapshotTree in this structure
+ * could be a partial snapshot tree consisting of group Ids for the missing children of the snapshotTree.
+ * @alpha
+ */
+export interface IPartialSnapshotWithContents {
+	snapshotTree: ISnapshotTree2;
+	blobsContents: Map<string, ArrayBuffer>;
+	ops: ISequencedDocumentMessage[];
+
+	/**
+	 * Sequence number of the snapshot
+	 */
+	sequenceNumber: number | undefined;
+
+	/**
+	 * Sequence number for the latest op/snapshot for the file.
+	 */
+	latestSequenceNumber: number | undefined;
```

Comment:
Maybe latest known?
How is this different from max(sequenceNumber, ops[ops.length-1]) (let's assume it picks the other value if one of them is undefiled)?
I.e. is it here for convenience, or it actually carries more information?
If would be great to capture that in comments, and also evaluate if we want to have it.

Reply:
We get this latestSequenceNumber from the service in snapshot and it is the latest seq number which is currently known to service(push may not have push latest ops yet). But it is not used in client code anywhere. So removing it.